src/lib/plugins/live-rows/live-rows.component.ts

```diff
@@ -0,0 +1,103 @@
+import { Component, OnInit, Input, ChangeDetectionStrategy } from '@angular/core';
+import { GridModel } from '../../plugins/plugin.service';
+import { PluginService } from '../plugin.service';
+import { GRID_PREFIX } from 'ng2-qgrid/core/definition';
+@Component({
+	selector: 'q-grid-live-rows',
+	template: '<ng-content></ng-content>',
+	providers: [PluginService],
+	changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class LiveRowsComponent implements OnInit {
+
+	@Input('duration') duration = 1000;
+
+	model: GridModel;
+	currentValues = [];
+	previousValues = [];
+	isAnimating = false;
+
+	constructor(private plugin: PluginService) {}
+
+	ngOnInit() {
+
+		const { model } = this.plugin;
+
+		model.animation({
+			apply: (memo, context, complete) => {
+
+				this.previousValues = this.currentValues || [];
+				this.currentValues = memo.rows;
+
+				if (!this.previousValues.length || !this.currentValues.length || this.isAnimating) {
+					complete(memo);
+					return;
+				}
+
+				this.isAnimating = true;
+
+				for (const i in this.previousValues) {
```

Comment:
please use for(const i = 0, length = this.prevRows.length; i < length...
it's really more readable to work with indicies like that